fixup! c++ / mojo changes for 'external window mode'

tonikitoo opened the pull request:
Another preparation work for the refactor patch
    
    Basically, it calls WindowTree::AddRoot from WindowTreeHostFactory
    rather than from Display::InitRootDisplay. This brings in no functionality
    change, but ensure we have a DefaultPlatformDisplay instance fully
    instantiated, something that is important in case we want to adopt the
    refactor branch.

msisov commented on services/ui/ws/display.cc:
```diff
@@ -476,7 +471,7 @@ void Display::OnFocusChanged(FocusControllerChangeSource change_source,
     embedded_tree_old = window_server_->GetTreeWithRoot(old_focused_window);
     if (embedded_tree_old) {
-      DCHECK_NE(owning_tree_old, embedded_tree_old);
+      // DCHECK_NE(owning_tree_old, embedded_tree_old);
```
nit: maybe add a comment here and below?

tonikitoo replied:
done

msisov commented on services/ui/ws/window_manager_display_root.cc:
```diff
@@ -25,9 +25,24 @@ WindowManagerDisplayRoot::WindowManagerDisplayRoot(Display* display)
-  root_.reset(window_server()->CreateServerWindow(
-      window_server()->display_manager()->GetAndAdvanceNextRootId(),
-      properties));
+  // We pretend the WindowManagerDisplayRoot::root_ is created by the root
```
maybe move this part to another fixup! c++ / mojo changes for 'external window mode' ?
I think it belongs to that part rather than deletion of windows.

tonikitoo replied:
I believe it belongs to its own fixup!, yes. the reason for leaving it separate, was for an easy revert, if we decide to.

msisov commented on services/ui/ws/window_tree.cc:
```diff
@@ -1364,6 +1365,28 @@ void WindowTree::DestroyWindows() {
+  // Hence its deletion will happen when WindowManagerState is gone.
+  if (window_server_->IsInExternalWindowMode()) {
+    for (auto& window_pair : created_window_map_copy) {
+      // During the main WindowTree shutdown, in external mode
```
this comment is not clear enough

tonikitoo replied:
I tried to clarify it. please recheck

tonikitoo committed:
fixup! c++ / mojo changes for 'external window mode'

Another preparation work for the refactor patch

Basically, it calls WindowTree::AddRoot from WindowTreeHostFactory
rather than from Display::InitRootDisplay. This brings in no functionality
change, but ensures we have a DefaultPlatformDisplay instance fully
instantiated when ::AddRoot is called, something that is important in
case we want to adopt the refactor branch.

tonikitoo commented:
@msisov please review again. this is another preparation PR.

msisov reviewed (approved):
lgtm
